src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.cs

```diff
@@ -121,32 +134,49 @@ public void ExecuteReturnOrTypeCommand(EditorCommandArgs args, Action nextHandle
                 return;
             }
 
-            var service = document.GetLanguageService<IEditorFormattingService>();
+            var service = document.GetLanguageService<IFormattingInteractionService>();
+            IEditorFormattingService? editorService = null;
             if (service == null)
             {
-                return;
+                editorService = document.GetLanguageService<IEditorFormattingService>();
+                if (editorService == null)
+                {
+                    return;
+                }
             }
 
             IList<TextChange>? textChanges;
 
             // save current caret position
             if (args is ReturnKeyCommandArgs)
             {
-                if (!service.SupportsFormatOnReturn)
+                if (!(service?.SupportsFormatOnReturn ?? editorService!.SupportsFormatOnReturn))
```

Comment:
don't shoot the messenger, but the appropriate thing to do here is wrap one of teh interfaces in a proxy object that gives you the other type.  that way the remainder of teh code deals with one type, and doesn't have this awful `?` then `!` shenanigans going on :(